tracer/src/Datadog.Trace/Iast/TaintedObjects.cs

```diff
@@ -24,7 +24,7 @@ public void TaintInputString(string stringToTaint, Source source)
             }
         }
 
-        public void Taint(object objectToTaint, Range[]? ranges)
+        public void Taint(object objectToTaint, Range[] ranges)
         {
             if (objectToTaint is not null && (objectToTaint is not string || objectToTaint as string != string.Empty))
```

Comment:
I think you can change this condition to not use multiple `is` and `as`

Reply:
I have updated the code.